### DISCUSSION TITLE
elf tools to update ELF file ( like objcopy ) ?

### DISCUSSION THREAD

**Discussion by sacnorthern** [eliben/pyelftools#561] [Q&A] [answered]
I found your ELF tools while looking for manipulation tools for our 32-bits MIPS and 64-bits Aarch64 ELF files.  Our flow is compile and build, then after linking insert two "signatures" into the ELF file, e.g. both SHA2-512.  A prior solution is a C program to read ELF, find the section, then insert a signature. Another solution uses a wrapper on binutil's **objcopy** program.

Reading the Python examples, they start with `open(elf, "rb")` call.  Its stated goal is "just like **readelf**".

So, does there exist "write_and_read_elf_tool" Python tool anywhere?  Has anyone asked this before and created some Python for it?  For us, we have old version of binutils, and `--update-section` was added after CentOS 6 was created ( yup, we installed Linux from a CD disc :) ).

As an aside, we noticed the **objcopy** will wipe out the "elf_header.machine" byte (offset 18) after using `--update-section` option.

thanks for your time and "wow" ELF tools!!
*brianW

**Comment by sevaa** [comment, marked as answer]
See https://github.com/eliben/pyelftools/discussions/502
